### ISSUE TITLE
MTE not working

### ISSUE THREAD

**Issue by Chipsum** [abbaye/WpfHexEditorControl#70]
**Describe the bug**
I used a custom TBL with MTE included, have a look :
https://pastebin.com/kJKqp4Sb
every character is working fine (DTE), but from line 231 to 235 I included MTE.
and this is what I got:
![image](https://user-images.githubusercontent.com/37517008/94325052-7ddb7b00-ff94-11ea-9bde-203d70d1cd71.png)

Meanwhile, it works perfectly on Windhex:
![image](https://user-images.githubusercontent.com/37517008/94325087-abc0bf80-ff94-11ea-8b0e-ee6d80f774b2.png)

**Comment by abbaye** [comment]
Can you send you TBL to derektremblay666@gmail.com please ?

I will check this weekend. :) 

All other MTE working ?

**Comment by abbaye** [comment]
Are you set the property TblShowMte = true ?

**Comment by Chipsum** [comment]
E-mail sent.
I am not sure about TblShowMe, I did not mess with the code yet , I did run the wpf example included only.

**Comment by abbaye** [comment]
If is a bug I will correct it 😁

Thank for submitting this issue

**Comment by abbaye** [comment]
I will check this today 🙂

**Comment by abbaye** [comment]
I'am working to find a solution but after i'm testing with other TBL and it's work fine... ⁉️

**Comment by Chipsum** [comment]
It's not working, the thing is , it does somehow display the MTE to MTE mapping, but instead , in the right text pannel , it takes the last byte of MTE and treats it as a dependent character.
For example let's consider the following MTE:
2241=[Line]
and let's say 41=A
In the text pannel it will be like
Sometext [LINE] A Sometext
In the picture below , it should be like:
[LB][LI3]THE FOX
Since [LI3] is a MTE mapped to 208D, it takes the last 8D and treats it as an M.
![image](https://user-images.githubusercontent.com/37517008/95514341-ac554f00-09b3-11eb-898f-7407dc1e7289.png)

**Comment by abbaye** [comment]
Yes I know but it's work in some's other TBL.

I will check again this weekend 🙂

**Comment by Chipsum** [comment]
This may help:
https://github.com/SourMesen/Mesen/blob/master/GUI.NET/Debugger/TblLoader.cs

**Comment by abbaye** [comment]
Thank you, I will check

The bug is not on my TBL loader but in when I call refresh () over control. 😬😟 I will correct this 💪

**Comment by Chipsum** [comment]
When you said that TBL loader is working , I had to check and seems alright
![image](https://user-images.githubusercontent.com/37517008/97139639-3bf05100-175b-11eb-98a5-cd59c2c9bea9.png)

It's now clear that it's not displaying it on the right text panel , meanwhile when I copy the text , it works fine.

**Comment by abbaye** [comment]
Yes you have totally reason. I don't find why it's not working with you TBL but working fine with mine.

I need to check for debugging this, this week

**Comment by abbaye** [comment]
Sorry for delay! university session take all my free time. It's just finish now. 

I will check this weekend to fix this

**Comment by Chipsum** [comment]
@abbaye 
Sorry I was busy at work too, wish you good luck with university.
Don't exhaust yourself, take your time (I insist :) )
Meanwhile, I will test it more.

**Comment by abbaye** [comment]
I think I have find the bug... I just need to sit and code the fix lol

**Comment by abbaye** [comment]
@Chipsum Hi mister! can you test my fix please.

on my side, it work fine with my Final Fantasy II TBL. 😄

**Comment by Chipsum** [comment]
@abbaye Hi !
I downloaded the latest (2.1.6) and built a clean one
But It doesn't seem solved
![image](https://user-images.githubusercontent.com/37517008/104732753-f467d880-573d-11eb-814c-ed53bebf0bb2.png)
Look at [END][L1], the MTE 2224 which is linked to [L1] isn't displayed compared to Windhex which is displayed correctly. 

Also, a bug came in the way , when I opened another file, this is what the first file became like:
![image](https://user-images.githubusercontent.com/37517008/104733034-62140480-573e-11eb-9a09-72c6e25a19b3.png)

**Comment by abbaye** [comment]
Hi mister

Can you sending your TBL and the ROM for a better testing

It's work fine with my file 🤨

**Comment by abbaye** [comment]
I will find this bug in priority.

**Comment by Chipsum** [comment]
@abbaye 
Sent :ok_hand:

**Comment by abbaye** [comment]
Check this my friend ! take the lastest code 😄 

Just moved one line of code 😅

![Screenshot_1](https://user-images.githubusercontent.com/17101371/105416343-faf7c380-5c07-11eb-8953-764a511d7e83.png)

**Comment by Chipsum** [comment]
Congrats ! 

I will check it as soon as possible :running:

**Comment by abbaye** [comment]
please come back

**Comment by Chipsum** [comment]
The main issue is solved 

![image](https://user-images.githubusercontent.com/37517008/105421129-e46e5e00-5c41-11eb-97ac-cd958842a094.png)

But Since the MTE is a 2-bytes it's still treating it (in the right-text panel) as a separated bytes by adding a space.

Meanwhile the feature of highlighting a byte seems to need to be fixed with MTE

For DTE it's fine
![image](https://user-images.githubusercontent.com/37517008/105421656-bb9a9880-5c42-11eb-83f7-ee75e803f5c7.png)

As you can see the mte ([END] which is hex (FF) works fine all three are highlighted in the image above.

But for MTE it's not, since it's highlighting technically only the 1st byte of an MTE

![image](https://user-images.githubusercontent.com/37517008/105421855-0d432300-5c43-11eb-8b31-fc07bfa82727.png)

But that's enough for now I think, since the main issue is fixed, I will open another issues next time :)

**Comment by abbaye** [comment]
Yes it's a little bug. I will correct this

**Comment by abbaye** [comment]
70-80% completed

**Comment by abbaye** [comment]
Are you checked the last code ? I think it's ok 😄
